filesystem/fat32/fat32.go

```diff
@@ -631,7 +631,8 @@ func (fs *FileSystem) writeDirectoryEntries(dir *Directory) error {
 	}
 	extraClusters := len(b)/(int(fs.bootSector.biosParameterBlock.dos331BPB.dos20BPB.sectorsPerCluster)*fs.bytesPerCluster) - len(clusterList)
 	if extraClusters > 0 {
-		clusters, err := fs.allocateSpace(uint64(extraClusters), clusterList[len(clusterList)-1])
+		extraBytes := extraClusters * fs.bytesPerCluster
+		clusters, err := fs.allocateSpace(uint64(fs.bytesPerCluster + extraBytes), clusterList[len(clusterList)-1])
```

Comment:
Agreed that the previous `allocateSpace(extraClusters` was wrong, since the first arg to `allocateSpace` is the file size. But then the above is incorrect too. You are just passing it the `bytesPerCluster` and the `extraBytes`. Shouldn't we just be passing it `len(b)`?

Reply:
if we pass len(b) then we would also have to pass the first cluster and extraClusters is not needed at all. I can switch to that again.

Reply:
Here is the comment on `allocateSpace()`:

```go
// allocateSpace ensure that a cluster chain exists to handle a file of a given size.
// arguments are file size in bytes and starting cluster of the chain
// if starting is 0, then we are not (re)sizing an existing chain but creating a new one
// returns the indexes of clusters to be used in order
```

The file is of a given size, and starting cluster of the chain. So we should stick to that.

Reply:
Fixed, needed a rebase to use the fixed `allocateSpace` method.